modules/stochastic_tools/src/surrogates/PolynomialRegressionSurrogate.C

```diff
@@ -36,19 +36,31 @@ PolynomialRegressionSurrogate::evaluate(const std::vector<Real> & x) const
               "Input point does not match dimensionality of training data.");
 
   Real val(0.0);
-  unsigned int n_terms(_power_matrix.size());
-
-  for (unsigned int i = 0; i < n_terms; ++i)
+  for (unsigned int i = 0; i < _power_matrix.size(); ++i)
   {
-    std::vector<unsigned int> i_powers(_power_matrix[i]);
-
     Real tmp_val(1.0);
-    for (unsigned int j = 0; j < i_powers.size(); ++j)
-    {
-      tmp_val *= pow(x[j], i_powers[j]);
-    }
-    val += _coeff[i] * tmp_val;
+    for (unsigned int j = 0; j < _power_matrix[i].size(); ++j)
+      tmp_val *= pow(x[j], _power_matrix[i][j]);
+    val += _coeff[0][i] * tmp_val;
   }
 
   return val;
 }
+
+void
+PolynomialRegressionSurrogate::evaluate(const std::vector<Real> & x, std::vector<Real> & y) const
+{
+  // Check whether input point has same dimensionality as training data
+  mooseAssert(_power_matrix[0].size() == x.size(),
+              "Input point does not match dimensionality of training data.");
+
+  y.assign(_coeff.size(), 0.0);
+  for (unsigned int i = 0; i < _power_matrix.size(); ++i)
+  {
+    Real tmp_val(1.0);
+    for (unsigned int j = 0; j < _power_matrix[i].size(); ++j)
+      tmp_val *= pow(x[j], _power_matrix[i][j]);
```

Comment:
Check out MathUtils::pow. If this is std::pow, its notoriously slow

Reply:
I'm not actually sure which `pow` this is calling. It's not `std::pow` I will update to be sure though.